### ISSUE TITLE
`& + .class` is invalid

### ISSUE THREAD

**Issue by joaomilho** [brigade/scss-lint#752]
I have the linter complaining about this:

```
.foo {
  & + .bar {
  }
}
```

And it's ok if I do
```
.foo {
  + .bar {
  }
}
```

From what I see in the tests, it doesn't cover this _specific_ case (only when you have & on both sides). Is this syntax ok for SASS?

```
  context 'when an ampersand precedes a sibling operator' do
    let(:scss) { <<-SCSS }
      p {
        & + & {}
        & ~ & {}
      }
    SCSS

    it { should_not report_lint }
  end
```

**Comment by abea** [comment]
If you're getting `UnnecessaryParentReference` on the first example, that's right. The ampersand is not needed there to achieve the same result. You shouldn't get the warning when you have two used like at the bottom.

**Comment by sds** [comment]
As @abea mentioned, `& + .bar` and `+ .bar` are equivalent, so the former is reported as a lint.

The same does not apply to selectors with two or more parent references, which is what the test in question is asserting.

Hope that helps!

**Comment by joaomilho** [comment]
Thanks!
